Enable liveWaitSupport, and play a stream before publish, a NullPointerException occur.

# Issue

### Short description
Enable liveWaitSupport, and play a stream before publish, a NullPointerException occur.
[ERROR] [NioProcessor-2] org.red5.server.messaging.InMemoryPullPullPipe - exception when pulling message from provider
java.lang.NullPointerException: null

### Environment

[] Operating system and version: CentOS 7
[] Java version: 1.8
[] Red5 version: 1.2.3


### Expected behavior
Play a stream before publish, then publish the stream, play should auto start.

### Actual behavior
Play a stream before publish, a NullPointerException occur.
After the stream published, play did not auto start.

### Steps to reproduce
1. Change conf/red5-common.xml property 'liveWaitSupport' to 'true'
2. Restart Red5 Server
3. Play a stream before publish.


### Logs
https://pastebin.com/dGtSdZ44

--- a/src/main/java/org/red5/server/stream/ProviderService.java
+++ b/src/main/java/org/red5/server/stream/ProviderService.java
@@ -218,6 +218,11 @@ public class ProviderService implements IProviderService {
                     log.debug("File {} not found, nulling it", filename);
                 }
             }
+            // check file existence
+            if (file != null && !file.exists()) {
+                // if it does not exist then null it out
+                file = null;
+            }
         } catch (IOException e) {
             log.info("Exception attempting to lookup file: {}", e.getMessage());
             if (log.isDebugEnabled()) {

